lib/connect.js

```diff
@@ -525,6 +525,16 @@ async function roundPunch (c, serverAddress, remoteToken, clientRelay) {
     if (!c.relayToken) throw REMOTE_NOT_HOLEPUNCHING()
   }
 
+  if (c.puncher.remoteFirewall >= FIREWALL.RANDOM || c.puncher.nat.firewall >= FIREWALL.RANDOM) {
+    if (c.dht._randomPunches >= c.dht._randomPunchLimit) {
+      if (!c.relayToken) throw HOLEPUNCH_ABORTED()
+      await c.sleeper.pause(5000 + Math.round(Math.random() * 5000))
```

Comment:
unsure if this is valid or if this is implicitly tied to the UDP session, will try to confirm